internal/app/machined/pkg/controllers/k8s/control_plane_static_pod.go

```diff
@@ -0,0 +1,435 @@
+				Name:      "kube-apiserver",
+				Namespace: "kube-system",
+				Labels: map[string]string{
+					"tier":            "control-plane",
+					"k8s-app":         "kube-apiserver",
+					"secrets-version": secretsVersion,
+				},
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:    "kube-apiserver",
+						Image:   cfg.Image,
+						Command: args,
+						Env: []v1.EnvVar{
+							{
+								Name: "POD_IP",
+								ValueFrom: &v1.EnvVarSource{
+									FieldRef: &v1.ObjectFieldSelector{
+										FieldPath: "status.podIP",
+									},
+								},
+							},
+						},
+						VolumeMounts: []v1.VolumeMount{
+							{
+								Name:      "secrets",
+								MountPath: constants.KubernetesAPIServerSecretsDir,
+								ReadOnly:  true,
+							},
+						},
+					},
+				},
+				HostNetwork: true,
+				SecurityContext: &v1.PodSecurityContext{
+					RunAsNonRoot: pointer.ToBool(true),
+					RunAsUser:    pointer.ToInt64(constants.KubernetesRunUser),
+				},
+				Volumes: []v1.Volume{
+					{
+						Name: "secrets",
+						VolumeSource: v1.VolumeSource{
+							HostPath: &v1.HostPathVolumeSource{
+								Path: constants.KubernetesAPIServerSecretsDir,
+							},
+						},
+					},
+				},
+			},
+		})
+
+		return nil
+	})
+}
+
+func (ctrl *ControlPlaneStaticPodController) manageControllerManager(ctx context.Context, r controller.Runtime,
+	logger *log.Logger, configResource *config.K8sControlPlane, secretsVersion string) error {
+	cfg := configResource.ControllerManager()
+
+	args := []string{
+		"/go-runner",
+		"/usr/local/bin/kube-controller-manager",
+		"--allocate-node-cidrs=true",
+		fmt.Sprintf("--cloud-provider=%s", cfg.CloudProvider),
+		fmt.Sprintf("--cluster-cidr=%s", cfg.PodCIDR),
+		fmt.Sprintf("--service-cluster-ip-range=%s", cfg.ServiceCIDR),
+		fmt.Sprintf("--cluster-signing-cert-file=%s", filepath.Join(constants.KubernetesControllerManagerSecretsDir, "ca.crt")),
+		fmt.Sprintf("--cluster-signing-key-file=%s", filepath.Join(constants.KubernetesControllerManagerSecretsDir, "ca.key")),
+		"--configure-cloud-routes=false",
+		fmt.Sprintf("--kubeconfig=%s", filepath.Join(constants.KubernetesControllerManagerSecretsDir, "kubeconfig")),
+		"--leader-elect=true",
+		fmt.Sprintf("--root-ca-file=%s", filepath.Join(constants.KubernetesControllerManagerSecretsDir, "ca.crt")),
+		fmt.Sprintf("--service-account-private-key-file=%s", filepath.Join(constants.KubernetesControllerManagerSecretsDir, "service-account.key")),
+		"--profiling=false",
+	}
+
+	for k, v := range cfg.ExtraArgs {
+		args = append(args, fmt.Sprintf("--%s=%s", k, v))
+	}
+
+	//nolint: dupl
+	return r.Update(ctx, k8s.NewStaticPod(k8s.ControlPlaneNamespaceName, "kube-controller-manager", nil), func(r resource.Resource) error {
+		r.(*k8s.StaticPod).SetPod(&v1.Pod{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "Pod",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "kube-controller-manager",
```

Comment:
I remember that bootkube adds node name to the static pod name. I guess that's to avoid conflicts with non-static ones when they start.
Here I guess they always run as static pods, right?